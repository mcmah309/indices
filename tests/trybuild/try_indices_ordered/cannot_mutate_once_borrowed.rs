use indices::try_indices_ordered;

fn main() {
    one();
    two();
    three();
    four();
    five();
}

fn one() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four,) = try_indices_ordered!(&mut slice, 4).unwrap();
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}");
}

fn two() {
    let mut slice = [0, 1, 2, 3, 4];
    let (one, four) = try_indices_ordered!(&mut slice, 1, 4).unwrap();
    slice[1] *= 2;
    *one *= 2;
    slice[1] *= 2;
    *one *= 2;
    println!("{one}, {four}");
}

fn three() {
    let mut slice = [0, 1, 2, 3, 4];
    let (one, two, four) = try_indices_ordered!(&mut slice, 1, 2, 4).unwrap();
    slice[1] *= 2;
    *one *= 2;
    slice[1] *= 2;
    *one *= 2;
    println!("{one}, {two}, {four}");
}

fn four() {
    let mut slice = [0, 1, 2, 3, 4];
    let (one, two, three, four) = try_indices_ordered!(&mut slice, 1, 2, 3, 4).unwrap();
    slice[1] *= 2;
    *one *= 2;
    slice[1] *= 2;
    *one *= 2;
    println!("{one}, {two}, {three}, {four}");
}

fn five() {
    let mut slice = [0, 1, 2, 3, 4];
    let (zero, one, two, three, four) = try_indices_ordered!(&mut slice, 0, 1, 2, 3, 4).unwrap();
    slice[0] *= 2;
    *zero *= 2;
    slice[0] *= 2;
    *zero *= 2;
    println!("{zero}, {one}, {two}, {three}, {four}");
}
