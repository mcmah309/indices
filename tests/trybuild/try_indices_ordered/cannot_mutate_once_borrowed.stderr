error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:14:5
   |
13 |     let (four,) = try_indices_ordered!(&mut slice, 4).unwrap();
   |                                        ---------- `slice` is borrowed here
14 |     slice[4] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
15 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:14:5
   |
13 |     let (four,) = try_indices_ordered!(&mut slice, 4).unwrap();
   |                                        ---------- `slice` is borrowed here
14 |     slice[4] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
15 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:24:5
   |
23 |     let (one, four) = try_indices_ordered!(&mut slice, 1, 4).unwrap();
   |                                            ---------- `slice` is borrowed here
24 |     slice[1] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
25 |     *one *= 2;
   |     --------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:24:5
   |
23 |     let (one, four) = try_indices_ordered!(&mut slice, 1, 4).unwrap();
   |                                            ---------- `slice` is borrowed here
24 |     slice[1] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
25 |     *one *= 2;
   |     --------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:34:5
   |
33 |     let (one, two, four) = try_indices_ordered!(&mut slice, 1, 2, 4).unwrap();
   |                                                 ---------- `slice` is borrowed here
34 |     slice[1] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
35 |     *one *= 2;
   |     --------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:34:5
   |
33 |     let (one, two, four) = try_indices_ordered!(&mut slice, 1, 2, 4).unwrap();
   |                                                 ---------- `slice` is borrowed here
34 |     slice[1] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
35 |     *one *= 2;
   |     --------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:44:5
   |
43 |     let (one, two, three, four) = try_indices_ordered!(&mut slice, 1, 2, 3, 4).unwrap();
   |                                                        ---------- `slice` is borrowed here
44 |     slice[1] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
45 |     *one *= 2;
   |     --------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:44:5
   |
43 |     let (one, two, three, four) = try_indices_ordered!(&mut slice, 1, 2, 3, 4).unwrap();
   |                                                        ---------- `slice` is borrowed here
44 |     slice[1] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
45 |     *one *= 2;
   |     --------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:54:5
   |
53 |     let (zero, one, two, three, four) = try_indices_ordered!(&mut slice, 0, 1, 2, 3, 4).unwrap();
   |                                                              ---------- `slice` is borrowed here
54 |     slice[0] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
55 |     *zero *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/try_indices_ordered/cannot_mutate_once_borrowed.rs:54:5
   |
53 |     let (zero, one, two, three, four) = try_indices_ordered!(&mut slice, 0, 1, 2, 3, 4).unwrap();
   |                                                              ---------- `slice` is borrowed here
54 |     slice[0] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
55 |     *zero *= 2;
   |     ---------- borrow later used here
